x/lockup/keeper/lock.go

```diff
@@ -0,0 +1,189 @@
+func (k Keeper) GetAccountUnlockableCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
+	return k.getCoinsFromIterator(ctx, k.AccountLockIteratorBeforeTime(ctx, addr, ctx.BlockTime()))
+}
+
+// GetAccountLockedCoins Returns a locked coins that can't be withdrawn
+func (k Keeper) GetAccountLockedCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
+	return k.getCoinsFromIterator(ctx, k.AccountLockIteratorAfterTime(ctx, addr, ctx.BlockTime()))
+}
+
+// GetAccountLockedPastTime Returns the total locks of an account whose unlock time is beyond timestamp
+func (k Keeper) GetAccountLockedPastTime(ctx sdk.Context, addr sdk.AccAddress, timestamp time.Time) []types.PeriodLock {
+	return k.getLocksFromIterator(ctx, k.AccountLockIteratorAfterTime(ctx, addr, timestamp))
+}
+
+// GetAccountUnlockedBeforeTime Returns the total unlocks of an account whose unlock time is before timestamp
+func (k Keeper) GetAccountUnlockedBeforeTime(ctx sdk.Context, addr sdk.AccAddress, timestamp time.Time) []types.PeriodLock {
+	return k.getLocksFromIterator(ctx, k.AccountLockIteratorBeforeTime(ctx, addr, timestamp))
+}
+
+// GetAccountLockedPastTimeDenom is equal to GetAccountLockedPastTime but denom specific
+func (k Keeper) GetAccountLockedPastTimeDenom(ctx sdk.Context, addr sdk.AccAddress, denom string, timestamp time.Time) []types.PeriodLock {
+	return k.getLocksFromIterator(ctx, k.AccountLockIteratorAfterTimeDenom(ctx, addr, denom, timestamp))
+}
+
+// GetAccountLockedLongerThanDuration Returns account locked with duration longer than specified
+func (k Keeper) GetAccountLockedLongerThanDuration(ctx sdk.Context, addr sdk.AccAddress, duration time.Duration) []types.PeriodLock {
+	return k.getLocksFromIterator(ctx, k.AccountLockIteratorLongerThanDuration(ctx, addr, duration))
+}
+
+// GetAccountLockedLongerThanDurationDenom Returns account locked with duration longer than specified with specific denom
+func (k Keeper) GetAccountLockedLongerThanDurationDenom(ctx sdk.Context, addr sdk.AccAddress, denom string, duration time.Duration) []types.PeriodLock {
+	return k.getLocksFromIterator(ctx, k.AccountLockIteratorLongerThanDurationDenom(ctx, addr, denom, duration))
+}
+
+// GetLockByID Returns lock from lockID
+func (k Keeper) GetLockByID(ctx sdk.Context, lockID uint64) (*types.PeriodLock, error) {
+	lock := types.PeriodLock{}
+	store := ctx.KVStore(k.storeKey)
+	lockKey := LockStoreKey(lockID)
+	if !store.Has(lockKey) {
+		return nil, fmt.Errorf("lock with ID %d does not exist", lockID)
+	}
+	bz := store.Get(lockKey)
+	k.cdc.MustUnmarshalJSON(bz, &lock)
+	return &lock, nil
+}
+
+// GetPeriodLocks Returns the period locks on pool
+func (k Keeper) GetPeriodLocks(ctx sdk.Context) ([]types.PeriodLock, error) {
+	return k.getLocksFromIterator(ctx, k.LockIterator(ctx)), nil
+}
+
+// UnlockAllUnlockableCoins Unlock all unlockable coins
+func (k Keeper) UnlockAllUnlockableCoins(ctx sdk.Context, account sdk.AccAddress) ([]types.PeriodLock, sdk.Coins, error) {
```

Comment:
It seems that the intention of this method is to unlock all unlockable coins from a specific account. But, the argument `account sdk.AccAddress` is not used and it may unlock all coins from everyone. Was the intention to use `AccountLockIteratorBeforeTime` and not `LockIteratorBeforeTime`?

Reply:
Thanks for finding this!